### ISSUE TITLE
Content integration test will start failing if BodyScannerComponent is refactored

### ISSUE THREAD

**Issue by DoctorBeard** [space-wizards/space-station-14#10273]
## Description
Currently the test EntityTest.AllComponentsOneEntityDeleteTest only seems to work because the BodyScannerComponent has `Owner.EnsureComponentWarn<ServerUserInterfaceComponent>();` in its Initialize. Once this line is removed the test will start to fail because as part of the test the AtmosMonitoringSystem will cause the AtmosDeviceSystem to raise the AtmosDeviceDisabledEvent. This event will then get handled by GasPressurePumpSystem, GasVolumePumpSystem, GasFilterSystem and GasMixerSystem, all of which will try to use the UserInterfaceSystem to call `TryCloseAll`. Since the BodyScannerComponent no longer ensures the ServerUserInterfaceComponent it now seems to be missing when this happens causing the Resolve in `TryCloseAll` to log an error and ultimately failing the test.

**Reproduction**
Remove the line `Owner.EnsureComponentWarn<ServerUserInterfaceComponent>();` from the BodyScannerComponent, then run AllComponentsOneEntityDeleteTest. It should fail on AtmosMonitor because of one of the Gas related systems mentioned above.

**Additional context**
I frankly don't know what the best way to fix this is but I've got two options that both worked. Either add HasComp checks on the above systems so they won't actually run into the UserInterfaceSystem or change the Resolve in TryCloseAll to not log on missing.

**Comment by lzk228** [comment]
the body refactor is no more, probably should be closed

**Comment by slarticodefast** [comment]
Yeah, none of the functions mentioned even exist anymore. This can be closed.
